src/fprime/common/models/serialize/serializable_type.py

```diff
@@ -27,83 +29,51 @@ class SerializableType(ValueType):
+
+    @classmethod
+    def validate(cls, val):
         """Validate this object including member list and values"""
-        # Blank member list does not validate
-        if not self.mem_list:
-            return
-        elif not isinstance(self.mem_list, list):
-            raise TypeMismatchException(list, self.mem_list)
-        for member_name, member_val, format_string, description in self.mem_list:
-            # Check each of these members for correct types
-            if not isinstance(member_name, str):
-                raise TypeMismatchException(str, type(member_name))
-            elif not isinstance(member_val, BaseType):
-                raise TypeMismatchException(BaseType, type(member_val))
-            elif not isinstance(format_string, str):
-                raise TypeMismatchException(str, type(format_string))
-            elif description is not None and not isinstance(description, str):
-                raise TypeMismatchException(str, type(description))
-        # When a value is set and is not empty we need to set the member properties
-        if not val:
+        if not cls.MEMBER_LIST:
             return
-        # If a value is supplied then check each value against the member list
-        for val_member, list_entry in zip(val, self.mem_list):
-            _, member_list_val, _, _ = list_entry
-            member_list_val.validate(
-                val_member
-            )  # Insure that the the val_member is consistent with the existing member
-
-    @property
-    def mem_list(self):
-        """Gets the member list"""
-        return self.__mem_list
-
-    @mem_list.setter
-    def mem_list(self, mem_list):
-        """
-        Sets the member list and validates against the current value.
-        """
-        self.__mem_list = mem_list
-        if mem_list is None:
-            self.validate(self.mem_list)
-
-    def serialize(self):
-        """Serializes the members of the serializable"""
-        if self.mem_list is None:
-            raise NotInitializedException(type(self))
-        return b"".join(
-            [member_val.serialize() for _, member_val, _, _ in self.mem_list]
-        )
-
-    def deserialize(self, data, offset):
-        """Deserialize the values of each of the members"""
-        new_member_list = []
-        for entry1, member_val, entry3, entry4 in self.mem_list:
-            cloned = copy.copy(member_val)
-            cloned.deserialize(data, offset)
-            new_member_list.append((entry1, cloned, entry3, entry4))
-            offset += member_val.getSize()
-        self.mem_list = new_member_list
+        # Ensure that the supplied value is a dictionary
+        if not isinstance(val, dict):
+            raise TypeMismatchException(dict, type(val))
+        # Now validate each field as defined via the value
+        for member_name, member_type, _, _ in cls.MEMBER_LIST:
+            member_val = val.get(member_name, None)
+            if not member_val:
```

Comment:
If member_val is a bool and False, then this expression will be true and incorrectly raise a MissingMemberExcpetion.